File: app/assets/javascripts/models/question_comment.js
_.constructor("QuestionComment", Model.Record, {
  constructorInitialize: function() {
    this.columns({
      questionId: 'key',
      creatorId: 'key',
      body: 'string',
      updatedAt: 'datetime',
      createdAt: 'datetime'
    });

    this.belongsTo('question');
    this.belongsTo('creator', {constructorName: "User"});
  },

  belongsToCurrentUser: function() {
    return this.creator() === Application.currentUser();
  },

  editableByCurrentUser: function() {
    return Application.currentUser().admin() || this.belongsToCurrentUser() || this.organization().currentUserIsOwner();
  },

  organization: function() {
    return this.question().organization();
  },

  formattedCreatedAt: function() {
    return $.PHPDate("n/j/y g:ia", this.createdAt());
  },

  trackCreate: function() {
    mpq.push(['track', "Create QuestionComment", {
      mp_note: this.body(),
      body: this.body(),
      id: this.id(),
      question: this.question().body(),
      questionId: this.questionId(),
      creator: this.creator().fullName(),
      creatorId: this.creatorId()
    }]);
  }
});

File: app/models/question_comment.rb
class QuestionComment < Prequel::Record
  column :id, :integer
  column :question_id, :integer
  column :creator_id, :integer
  column :body, :string
  column :created_at, :datetime
  column :updated_at, :datetime

  belongs_to :creator, :class_name => "User"
  belongs_to :question

  delegate :broadcast_channels, :to => :question, :allow_nil => true

  validates_presence_of :body

  def before_create
    self.creator = current_user
  end
end
